### PR TITLE
Approve check

### PR DESCRIPTION
**Pull request by hackaugusto**
Review after: #1123

### CODE REVIEW

**Comment by hackaugusto** [comment]
> Use single quotes please, good to merge afterwards.

I'm using double quotes to avoid escaping `account's ` and to keep the strings consistent, should I use:

```python
                msg = (
                    'Approve failed. \n'
                    "Your account's balance is 0 (zero), either the smart "
                    'contract is not a valid ERC20 token or you don't have funds '
                    'to use for openning a channel. '
                )
```

or

```python
                msg = (
                    'Approve failed. \n'
                    'Your account\'s balance is 0 (zero), either the smart '
                    'contract is not a valid ERC20 token or you don't have funds '
                    'to use for openning a channel. '
                )
```

? (we have not discussed such edge case)

**Comment by palango** [comment]
I like the second option more, seems clearer overall.

**Comment by hackaugusto** [comment]
Then we need to update our coding guidelines, please ask for the other opinions and let's settle with a single style.

**Comment by palango** [comment]
We need to figure this out, but after reading again I think it should be `Your account balance is zero`, circumventing the whole problem \o/

**Comment by palango** [comment]
Fine with me. But still think it should be `Your account balance is 0` here.

**Comment by hackaugusto** [comment]
changed the wording

File: raiden/network/proxies/token.py [hunk 1]
```diff
@@ -71,8 +71,43 @@ def approve(self, contract_address, allowance):
...
+                raise TransactionThrew(msg, receipt_or_none)
+
+            # The approve call failed, check the user has enough balance
+            # (assuming the token smart contract may check for the maximum
+            # allowance, which is not necessarily the case)
+            elif user_balance < allowance:
+                msg = (
+                    'Approve failed. \n'
+                    'Your account balance is {}, nevertheless the call to '
+                    ' approve failed. Please make sure the corresponding smart '
```
Comment (palango) [inline raiden/network/proxies/token.py]: Still a space to much here and next line

Comment (hackaugusto) [reply raiden/network/proxies/token.py]: fixed

File: raiden/network/proxies/token.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def approve(self, contract_address, allowance):
             if user_balance == 0:
                 msg = (
                     "Approve failed. \n"
-                    "Your account's balance is 0 (zero), either the smart "
+                    "Your account balance is 0 (zero), either the smart "
```
Comment (palango) [inline raiden/network/proxies/token.py]: single quotes

File: raiden/network/proxies/token.py [hunk 3]
```diff
@@ -91,21 +91,21 @@ def approve(self, contract_address, allowance):
...
             elif user_balance < allowance:
                 msg = (
-                    "Approve failed. \n"
-                    "Your account's balance is {}, nevertheless the call to "
-                    " approve failed. Please make sure the corresponding smart "
-                    " contract is a valid ERC20 token."
+                    'Approve failed. \n'
+                    'Your account balance is {}, nevertheless the call to '
+                    'approve failed. Please make sure the corresponding smart '
+                    ' contract is a valid ERC20 token.'
```
Comment (palango) [inline raiden/network/proxies/token.py]: missed this space

**Comment by palango** [comment]
Sorry Augusto, I'm being annoying :/

**Comment by palango** [comment]
Ready to merge once Travis gives green light.

**Pull request merged**